src/tracks/InstrumentTrack.cpp

```diff
@@ -249,6 +263,24 @@ MidiEvent InstrumentTrack::applyMasterKey( const MidiEvent& event )
 
 
 
+void InstrumentTrack::processCCEvent(int controller)
+{
+	// Does nothing if the LED is disabled
+	if( !m_midiCCEnable->value() )
+	{
+		return;
+	}
+
+	uint8_t channel = static_cast<uint8_t>( midiPort()->realOutputChannel() );
+	uint16_t cc = static_cast<uint16_t>( controller );
+	uint16_t value = static_cast<uint16_t>( m_midiCCModel[ controller ]->value() );
+
+	processOutEvent( MidiEvent( MidiControlChange, channel, cc, value ) );
```

Comment:
Both `processInEvent` and `processOutEvent` passes the MIDI events to the instruments, but each of them has their unique functionality:
- `processInEvent`: handle some MIDI events which aren't handled anywhere else
- `processOutEvent`: send MIDI events to external MIDI devices

I don't know your intention, but make sure you're using the correct one.

Reply:
I was reading `processInEvent` and `processOutEvents` again and I think I wasn't that off in understanding what they do. It can get a little bit confusing at some points though. Here are my thoughts:

Apparently, `processInEvent` processes events that are received through the MIDI Input port. That includes everything external sending events to the instrument track (either other tracks, MIDI keyboards/controllers or the piano widget itself). Some of those events will be processed in a particular way (i.e.: `MidiNoteOn` will make the instrument track create a `NotePlayHandle` for the note, etc), and most of them (but not all) will also trigger a `processOutEvent`, which will send the event further down the MIDI ports chain. MIDI Control Change events are not among those that trigger a `processOutEvent`. What that means is that the instrument plugin will receive the MIDI CC event through the `handleMidiEvent` method, but if you select another track on the MIDI Output dialog it will not receive those messages.

`processOutEvent` basically processes events generated by the instrument track that will be sent out through the MIDI Out ports. All tracks that are connected to this track through MIDI Input/Output ports will receive messages processed through `processOutEvent` (that are not blocked by the MIDI output channel filter of course).

So the decision between both of them resides on how do we want to see those MIDI CC events being interpreted as: Are they supposed to be an emulation of a hardware MIDI controller and behave **exactly** like we were using a physical one or should they be thought as events generated by the instrument track itself? I didn't have enough time to weight pros/cons of each approach so I left the code as is and added a TODO comment so I come back to this. Honestly, as I write this I think emulating a physical hardware makes sense (using `processInEvent`), though then some knobs are interpreted in a way that might not be obvious to the user (Knob 64 will enable/disable sustain and Knobs 120/123/124/125/126/127 will silence all notes). Still sounds like a good idea, since the same would happen with a physical MIDI controller.